modelling/src/main/java/org/axonframework/modelling/command/AggregateAnnotationCommandHandler.java

```diff
@@ -387,17 +418,19 @@ public boolean canHandle(CommandMessage<?> message) {
     private class AlwaysCreateAggregateCommandHandler implements MessageHandler<CommandMessage<?>> {
 
         private final MessageHandlingMember<? super T> handler;
-        private final Callable<T> factoryMethod;
+        private final Function<Object, T> factoryMethod;
 
         private AlwaysCreateAggregateCommandHandler(MessageHandlingMember<? super T> handler,
-                                                    Callable<T> factoryMethod) {
+                                                    Function<Object, T> factoryMethod) {
             this.handler = handler;
             this.factoryMethod = factoryMethod;
         }
 
         @Override
         public Object handle(CommandMessage<?> command) throws Exception {
-            Aggregate<T> aggregate = repository.newInstance(factoryMethod);
+            VersionedAggregateIdentifier commandMessageVersionedId = commandTargetResolver.resolveTarget(command);
```

Comment:
That's a good point, actually, for both cases, it might happen that there is no Identifier present in the command message. In that case, we'd provide a `null` to the factory. 

However, due to the API of both `CommandTargetResolver` and `VersionedAggregateIdentifier` we can not allow nulls here. 

What we can do is introduce another `default` method on the Resolver, something along the lines of `resolveTargetOrNull` to return a null instead of a `VersionedAggregateIdentifier` instance.

Reply:
We've had some out-of-GitHub-scope discussions about this subject, where we came to the following conclusion.

We concluded that from the perspective of the command dispatcher, it comes down to whether it is clear if the command message can create or update something.
This question arose as the dispatcher isn't clear on the exact implementation of the command handler.
Thus it wouldn't know whether the command acts as a create, update, or create-or-update.

The command API, aka the payload of the `CommandMessage`, can be clear about this though.
By making the identifier field nullable or required, for example.

With this in mind, I think the adjustment that the `CreationPolicyAggregateFactory` allows a `null` identifier is correct.
Furthermore, the resolver should be able to return a `null` identifier as well.